rest-api/tools/import_organizations.py

```diff
@@ -139,14 +139,29 @@ def _cleanup_old_entities(self, session, row_list, dry_run):
       hpo_id_list = []
       for hpo in hpos_to_remove:
         old_hpo = self.hpo_dao.get_by_name(hpo)
-        if old_hpo:
-          log_prefix = '(dry run) ' if dry_run else ''
-          logging.info(log_prefix + 'Deleting old HPO no longer in Google sheet: %s', old_hpo.name)
+        if old_hpo and old_hpo.isObsolete != ObsoleteStatus.OBSOLETE:
           hpo_id_list.append(old_hpo.hpoId)
           self.deletion_count += 1
+        elif old_hpo.isObsolete == ObsoleteStatus.OBSOLETE:
+          logging.info('Not attempting to delete hpo [%s] with existing obsolete status',
+                       old_hpo.name)
 
       if hpo_id_list and not dry_run:
-        self.delete_sql_statement(session, hpo_id_list)
+        logging.info(log_prefix + 'Marking old HPO as obsolete referenced in other table: %s',
+                     old_hpo.name)
+        str_list = ','.join([str(i) for i in hpo_id_list])
+
+        sql = """ UPDATE HPO
+            SET is_obsolete = 1
+            WHERE hpo_id in ({params})""".format(params=str_list)
+
+        session.execute(sql)
+
+        self.hpo_dao._invalidate_cache()
```

Comment:
Might want to do this after delete_sql_statement?

Reply:
done